rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala

```diff
@@ -171,11 +171,27 @@ private class Translator {
     combineTree(lazyExprs, ring)
   }
 
+  private def kahanSumUpdate(accumulator: Expr,
+                             compensation: Expr,
+                             term: Expr) = {
+    val nextExpr = binaryExpr(term, compensation, SubtractOp)
+    val newAccumulator = binaryExpr(accumulator, nextExpr, AddOp)
```

Comment:
With `ref(accumulator)` here, I get an uninitialized register error. As is, even with these other `refs` I still get hanging GC.

Reply:
Have you done graphviz plots to visualize what's going on?